lib/optimize.ts

```diff
@@ -523,6 +523,13 @@ function optimizeExpression(expression : Ast.Expression, allow_projection=true)
         return expression;
     }
 
+    if (expression instanceof Ast.JoinExpression) {
+        const lhs = optimizeExpression(expression.lhs);
+        const rhs = optimizeExpression(expression.rhs);
+        const condition = expression.condition ? optimizeFilter(expression.condition) : null;
```

Comment:
The order of the operands matter for the join condition. Are you sure you can optimize blindly?
Maybe add a test case?

Reply:
because `first` comes before `second` so it should be fine (assuming we are keeping the prefix in the condition, if that is removed, yeah, then I need some special handling.